packages/core-data/src/fetch/__experimental-fetch-url-data.js

```diff
@@ -26,25 +32,45 @@ const CACHE = new Map();
  * @param {Object?} options any options to pass to the underlying fetch.
  * @example
  * ```js
- * import { __experimentalFetchRemoteUrlData as fetchRemoteUrlData } from '@wordpress/core-data';
+ * import { __experimentalFetchUrlData as fetchUrlData } from '@wordpress/core-data';
  *
  * //...
  *
  * export function initialize( id, settings ) {
  *
- * settings.__experimentalFetchRemoteUrlData = (
+ * settings.__experimentalFetchUrlData = (
  * url
- * ) => fetchRemoteUrlData( url );
+ * ) => fetchUrlData( url );
  * ```
  * @return {Promise< WPRemoteUrlData[] >} Remote URL data.
  */
-const fetchRemoteUrlData = async ( url, options = {} ) => {
+const fetchUrlData = async ( url, options = {} ) => {
 	const endpoint = '/__experimental/url-details';
 
 	const args = {
 		url: prependHTTP( url ),
 	};
 
+	if ( ! isURL( url ) ) {
+		return Promise.reject(
+			new TypeError( `${ url } is not a valid URL.` )
+		);
+	}
+
+	// Test for "http" based URL as it is possible for valid
+	// yet unusable URLs such as `tel:123456` to be passed.
+	const protocol = getProtocol( url );
+
+	if (
+		! isValidProtocol( protocol ) ||
+		! protocol.startsWith( 'http' ) ||
+		! /^https?:\/\/[^\/\s]/i.test( url )
+	) {
+		return Promise.reject(
+			new TypeError( `${ url } does not have a valid protocol.` )
+		);
+	}
+
```

Comment:
Moving all the previous URL checking logic into the main function rather than in the intermediary.